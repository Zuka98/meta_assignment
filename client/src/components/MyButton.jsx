import React from 'react';
import Button from '@material-ui/core/Button';

/* 
  Button Component
  @param {string} name
  @param {func} handleClick
  @returns button element
 */
export default function MyButton(props){
  const btn_name = 'btn' + props.name;
  return (
    <Button value={btn_name} variant="outlined" color="primary" onClick={e => props.handleClick(btn_name)}>
          {props.name}
    </Button>
  );
}