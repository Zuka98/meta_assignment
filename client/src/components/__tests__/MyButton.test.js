import React from 'react';
import { render, screen } from '@testing-library/react';
import { configure, shallow } from 'enzyme';
import MyButton from '../MyButton.jsx';
import Adapter from 'enzyme-adapter-react-16.1';

configure({adapter: new Adapter()});

describe('MyButton', () => {
  const mockFn = jest.fn();
  it('Should be defined', () => {
    expect(MyButton).toBeDefined();
  });
  it('Should render correctly', () => {
    const tree = shallow(
      <MyButton name='1' handleClick={mockFn} />
    );
    expect(tree).toMatchSnapshot();
  });
  it('Should have a button value', () => {
    const tree = shallow(
      <MyButton name='1' handleClick={mockFn} />
    );
    expect(typeof(tree.getElement('button').props.value)).toBe('string');
    expect(tree.getElement('button').props.value).toEqual('btn1');
  });
  it('Should have a click function', () => {
    const tree = shallow(
      <MyButton name='1' handleClick={mockFn} />
    );
    tree.simulate('click');
    expect(mockFn).toHaveBeenCalled();
  });
});
