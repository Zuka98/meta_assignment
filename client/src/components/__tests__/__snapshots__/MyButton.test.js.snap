// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyButton Should render correctly 1`] = `ShallowWrapper {}`;

exports[`MyButton should render correctly 1`] = `ShallowWrapper {}`;
